cmd/cstor-sidecar/ccrd/mainspc.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ccrd
+
+import (
+	"time"
+
+	"github.com/golang/glog"
+	kubeinformers "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	// Uncomment the following line to load the gcp plugin (only required to authenticate against GKE clusters).
+	// _ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	
+	clientset "github.com/openebs/maya/pkg/client/clientset/versioned"
+	informers "github.com/openebs/maya/pkg/client/informers/externalversions"
+	"github.com/openebs/maya/pkg/signals"
+)
+
+func Checkcrd(kuberconfig string) {
+	masterURL := ""
+	// set up signals so we handle the first shutdown signal gracefully
+	stopCh := signals.SetupSignalHandler()
+
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		glog.Errorf("failed to get k8s Incluster config. %+v", err)
+		if kuberconfig == "" {
+			glog.Fatalf("kubeconfig is empty")
+		} else {
+			cfg, err = clientcmd.BuildConfigFromFlags(masterURL, kuberconfig)
```

Comment:
It is better to check for presence of kuberconfig , if available then invoke the BuildConfigFromFlags. If not available then make use of rest.InClusterConfig().